src/Microsoft.AspNet.OData.Shared/DeltaSetOfT.cs

```
@@ -45,5 +50,386 @@ protected override void InsertItem(int index, IDeltaSetItem item)
+                {
+                    //{
+                    //    "@odata.context":"http://localhost:6285/odata/$metadata#Customers/$delta",
+                    //    "value":[
+                    //        {
+                    //            "@odata.type":"CustomersApiEF.Models.Customer",
+                    //            "Id":1,
+                    //            "Name":"Customer1",
+                    //            "Orders@odata.delta":[
+                    //                {
+                    //                    "@odata.id":"Customers(3)/Orders(1005)",
+                    //                    "Quantity": 1005
+                    //                },
+                    //                {
+                    //                    "Id": 2000,
+                    //                    "Price":200,
+                    //                    "Quantity":90
+                    //                }
+                    //            ]
+                    //        }
+                    //    ]
+                    //}
+
+                    // If we have a request payload above and we are handling the changed values in Orders,
+                    // The apiHandlerOfT is OrdersAPIHandler with parent as Customers(1).
+                    // The object with "@odata.id" will have OrdersAPIHandler with parent as Customers(3).
+                    // The object with id 2000 will have OrdersAPIHandler similar to apiHandlerOfT.
+                    // The codebelow ensures we use the correct handler.
+                    // NOTE: Handler when we have odata.id will always be different than the parent handler.
+                    // However this is true if handlers are applied correctly.
+
+                    ODataAPIHandler<TStructuralType> odataPathApiHandler = apiHandlerFactory.GetHandler(changedObj.ODataPath) as ODataAPIHandler<TStructuralType>;
+
+                    if (odataPathApiHandler != null && changedObj.ODataPath.Any())
```

What happens if this condition is not met? Should we throw an exception?

---

This condition is only met when we have an `@odata.id`, If there isn't, we use the handler passed in the method.
So we shouldn't throw an exception

---

Aah I see. So this should only apply if we have an ODATAiD..